ethers-providers/src/transports/quorum.rs

```diff
@@ -179,6 +179,36 @@ impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
             .ok_or_else(|| ProviderError::CustomError("No Providers".to_string()))
     }
 
+    /// Returns the block height that a _quorum_ of providers have reached.
+    async fn get_quorum_block_number(&self) -> Result<U64, ProviderError> {
+        // To be even more robust, we could use an iterator and handle as futures respond to be
+        // tolerant to a provider that has stopped responding.
+        let mut numbers = join_all(self.providers.iter().map(|provider| async move {
+            let block = provider.inner.request("eth_blockNumber", QuorumParams::Zst).await?;
+            serde_json::from_value::<U64>(block).map(|b| (provider, b)).map_err(ProviderError::from)
+        }))
+        .await
+        .into_iter()
+        .sorted_by(|(_, block_a), (_, block_b)| {
+            // order by descending block number
+            block_a.cmp(block_b).reverse()
+        })
+        .collect::<Result<impl Iterator<Item = _>, _>>()?;
+
+        // find the highest possible block number a quorum agrees on
+        let mut weight_set = vec![];
+        let mut block = U64::from(0);
+        for (provider, n) in numbers {
+            weight_set.push(provider);
+            debug_assert!(block == 0 || block > n);
```

Comment:
should this be `debug_assert!(block == 0 || block >= n);` ?

Reply:
Yes, it should, good catch